Sleeping to wait for the winrm password to be available.

alexlod opened the pull request:
Disclaimer: I've never built a Vagrant plugin and am not familiar with the Vagrant plugin API.

alexlod committed:
Sleeping to wait for the winrm password to be available.

alexlod commented:
Here's a screen shot of the EC2 console explaining why this is needed:

![screen shot 2016-10-26 at 1 28 59 pm](https://cloud.githubusercontent.com/assets/39045/19743963/4144f76a-9b80-11e6-8826-cc55f1ab7319.png)

rafd123 commented:
Thanks for the due diligence and the pull request, Alex!

In order to make this a robust solution, we should be trying to fetch the password in a loop. One minute may not be enough.

It turns out the WinRM communicator `ready?` method is already wrapped by a retry loop that I think we can take advantage of.

Seems like simply raising a `VagrantPlugins::CommunicatorWinRM::Errors::TransientError` if `password_data` is `nil` it should do the trick; the outer retry loop will do the rest. See https://github.com/mitchellh/vagrant/blob/master/plugins/communicators/winrm/communicator.rb#L103-L122

Another advantage of this approach is that the time it takes for the password to become available will get figured into the overall timeout that the user configures to wait for WinRM to become available.

alexlod committed:
Responding to @rafd123's feedback.

alexlod commented:
Great suggestion, @rafd123, and thanks for going into detail and making it easy on me.

I just pushed a tested, new commit that addresses your feedback.

Let me know if you'd like me to rebase the commits into a single commit.

matt-richardson commented:
Just tested this commit (via copy and paste), as I hit this exact error, and it worked well.

Looking forward to the full release/

rafd123 commented:
@alexlod I took the liberty of rebasing and updating the commit message.

I also felt like the `ui.info` message needed to be moved and modified in order to prevent the repetitive "Getting WinRM password from AWS..." message as we kept retrying.

See https://github.com/rafd123/vagrant-aws-winrm/commit/ea70b3976676fcf40487d31a687b437631ead84a.

Finally, I went ahead and published the new gem.

Thanks for your help!

alexlod commented:
Awesome, thanks for the quick turnaround, @rafd123!
